Update draft-birkholz-scitt-software-use-cases.md (#1)
kkarunakaran89/draft-birkholz-scitt-software-supply-chain-use-cases: draft-birkholz-scitt-software-use-cases.md

## Firmware Delivery to large set of constrained IoT Devices

Firmware is a critical component for successful execution of any constrained IoT device. It is often the bedrock on which the security story of the devices it powers. For example, personal health monitoring devices (eHealth devices) are generally battery driven and offer health telemetry monitoring, such as temperature, blood pressure, and pulse rate. These devices typically connect to the Internet through an intermediary base station using wireless technologies. Through this connection, the telemetry data and analytics transfer, and devices receive firmware updates when published by the vendor. The public network, open distribution system, and firmware update process create several security challenges.

> Consumer of a firmware update system wants:

Author: Change to consumers and other interested parties

Personal health monitoring devices, i.e., eHealth devices, are generally battery driven and offer health telemetry monitoring, such as temperature, blood pressure, and pulse rate. These devices typically connect to the Internet through an intermediary base station using wireless technologies. Through this connection, the telemetry data and analytics transfer, and devices receive firmware updates when published by the vendor. The public network, open distribution system, and firmware update process create several security challenges.
* to know that the received firmware for system update is not faulty or malicious
* to know if the signing identity used to assert the authenticity of the firmware is somehow used to sign unintended updates
* to ascertain that the released firmware is not subverted or compromised due to an insider risk - be it malicious or otherwise
* to confirm that the publishers know if their deliverable has been compromised. Can they trust their key protection or audit logging?
* to know how the update client on an instance of a health monitoring system discerns a general update from one specially crafted for just a small subset of a fleet of devices

> Today, the best-in-class firmware vendors who supply the firmware also provide an update framework, which verifies the integrity and authenticity of firmware updates before allowing installation.

Reviewer: It looks like this sentence is lost from the paragraph at line 214. Recommend adding.

Author: Addressed in line 226 (there is no standardized way to: provide an update framework which allows verification of integrity and authenticity of firmware updates before installation)

> * reliabily discern an update that has been signed by the appropriate and intended signing identity

Author: change reliabily to reliably

* clearly communicate firmware revisioning and security checks completed to ensure the firmware is safe for install

## Software Integrator assembling a software product for a smart car

Software Integration is a complex activity. This typically involves getting various software components from multiple suppliers and producing an integrated package deployed as part of device assembly. For example, car manufacturers source integrated software for their autonomous vehicles from third parties that integrates software components from various sources. Integration complexity creates a higher risk of security vulnerabilities to the delivered software.

Consumer of an integrated software wants:

> * Bill of material capturing all software components used, with the SBOM stored in a secure tamperproof location

Reviewer: Suggest:
* to receive an alert when a vulnerability scan detects a known security issue on a running software component.

> * Bill of material capturing all software components used, with the SBOM stored in a secure tamperproof location

Author: to list all components present in software product, and identify and retrieve them from a secure and tamperproof location

* Verifiable proofs on build process and build environment with all supplier tiers to ensure end to end build quality and security

> * Vulnerability scan reports and approved malware mitigation plan before integration

Author: addressed by Kay's comment - remove 238

There is no standardized way to:

> * Provide SBOMs in a tamperproof ledger for customer consumption

Reviewer: Suggest: Notify software integrators of vulnerabilities identified during security scans of running software.

> * Provide SBOMs in a tamperproof ledger for customer consumption

Author: verify the statements/assertions made by an ISV regarding build process and build environment

Author: This is captured in essence in line 242

> * Provide valid attestations on build integrity to ensure conformance

Author: change attestation to statements/annotations

Consumers and other interested parties of a firmware update system wants:

* to know that the received firmware for system update is not faulty or malicious
* to know if the signing identity used to assert the authenticity of the firmware is somehow used to sign unintended updates
* to ascertain that the released firmware is not subverted or compromised due to an insider risk - be it malicious or otherwise
* to confirm that the publishers know if their deliverable has been compromised. Can they trust their key protection or audit logging?
* to know how the update client on an instance of a health monitoring system discerns a general update from one specially crafted for just a small subset of a fleet of devices

There is no standardized way to:

> * provide an update framework which allows verification of integrity and authenticity of firmware updates before installation

Reviewer 2: ```suggestion

* to verify that the firmware update seen by a single device, is indeed the same as seen by all the devices

```

> * provide an update framework which allows verification of integrity and authenticity of firmware updates before installation

Reviewer 2: ```suggestion
* provide an update framework that allows validation of authenticity of firmware revisions
* to verify that the firmware update seen by a single device, is indeed the same as seen by all the devices.
```

* reliably discern an update that has been signed by the appropriate and intended signing identity

> * clearly communicate firmware revisioning and security checks completed to ensure the firmware is safe for install

Reviewer 2: ```suggestion
* Make an informed judgement on all available information about firmware at the install time. For example, the firmware is still in a good state or otherwise?
```

## Software Integrator assembling a software product for a smart car

Software Integration is a complex activity. This typically involves getting various software components from multiple suppliers and producing an integrated package deployed as part of device assembly. For example, car manufacturers source integrated software for their autonomous vehicles from third parties that integrates software components from various sources. Integration complexity creates a higher risk of security vulnerabilities to the delivered software.

Consumer of an integrated software wants:

> * all components presents in a software product listed, and the ability to identify and retrieve them from a secure and tamperproof location

Reviewer 2: ```suggestion

```
